docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,640 @@
+
+To set common request headers in `HttpRequestMessage`, you can use the `Headers` property of the `HttpRequestMessage` object. This property provides access to the `HttpRequestHeaders` collection, where you can add or modify headers as needed.
+
+**Setting Common Default Request Headers in HttpClient**
+
+```csharp
+using System.Net.Http.Headers;
+
+var client = new HttpClient();
+client.DefaultRequestHeaders.Authorization =  new AuthenticationHeaderValue("Bearer", "token");
+```
+
+**Setting Common Request Headers in HttpRequestMessage**
+
+```csharp
+using System.Net.Http.Headers;
+
+var request = new HttpRequestMessage(HttpMethod.Get, uri);
+request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", "token");
+```
+
+### Example: Set Content Headers
+
+Content headers are used to provide additional information about the body of an HTTP request or response. When working with `HttpClient` in .NET, you can set content headers to specify the media type, encoding, and other metadata related to the content being sent or received. Properly configuring content headers ensures that the server and client can correctly interpret and process the content.
+
+```csharp
+var client = new HttpClient();
+var request = new HttpRequestMessage(HttpMethod.Post, uri);
+
+// Create the content and set the content headers
+var jsonData = "{\"key\":\"value\"}";
+var content = new StringContent(jsonData, Encoding.UTF8, "application/json");
+// This is also corresponding example to set `MediaType` in `HttpWebRequest`.
+content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+content.Headers.ContentEncoding.Add("utf-8");
+// WARNING: This is not needed in most of the cases, because most `HttpContent` implementations will calculate this field automatically.
+content.Headers.ContentLength = jsonData.Length;
+
+// Assign the content to the request
+request.Content = content;
+
+using var response = await client.SendAsync(request);
+```
+
+### Example: Set MaximumErrorResponseLength in HttpClient
+
+The `MaximumErrorResponseLength` usage allows developers to specify the maximum length of the error response content that the handler will buffer. This is useful for controlling the amount of data that is read and stored in memory when an error response is received from the server. By using this technique, you can prevent excessive memory usage and improve the performance of your application when handling large error responses.
+
+There are couple of ways to do that, we'll examine `TruncatedReadStream` technique on this example:
+
+:::code source="../snippets/httpclient/Program.TruncatedReadStream.cs" id="TruncatedReadStream":::
+
+And usage example of `TruncatedReadStream`:
+
+:::code source="../snippets/httpclient/Program.TruncatedReadStream.cs" id="TruncatedReadStreamUsage":::
+
+### Example: Apply CachePolicy Headers
```

Comment:
We may want to delete this section/example as it's a bit misleading.
With WebRequest, you would set these settings to control how the framework itself handles the caching of responses for you, not just to set headers.
HttpClient does no such caching, so just setting the headers isn't achieving anything on its own.

Reply:
We might add another explanation about how this actually should work normally, leave this as-is, so people can get the idea about how these headers can be set.